File: app/src/main/java/com/example/ogan/codedenim/gson/ApiMethods.java
package com.example.ogan.codedenim.gson;

import com.example.ogan.codedenim.courses.CourseRegister;
import com.example.ogan.codedenim.user.Corper;
import com.example.ogan.codedenim.user.Student;
import com.example.ogan.codedenim.user.User;

import java.util.ArrayList;

import okhttp3.ResponseBody;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface ApiMethods {

    @POST("api/accountapi/registercorper")
    Call<ResponseBody> registerCorper(@Body Corper corper);

    @POST("api/AccountApi/RegisterUnderGraduate")
    Call<ResponseBody> registerStudent(@Body Student student);

    @POST("api/AccountApi/OtherStudent")
    Call<ResponseBody> registerOtherStudent(@Body User user);

    @FormUrlEncoded
    @POST("token")
    Call<ResponseBody> loginCorper(@Field("username") String username, @Field("password") String password, @Field("grant_type") String grant_type);

    @GET("api/CourseCategories/CourseCategorys")
    Call<ArrayList<LearningPath>> getCategories(@Query("email") String email);

    @GET("api/Coursecategories/CourseCategory")
    Call<ArrayList<Course>> getCourses(@Query("id") int id);

    @GET("api/CourseCategories/MyCourses")
    Call<ArrayList<LearningPath>> getCourses(@Query("email") String email);

    @GET("api/Courses")
    Call<ArrayList<Course>> getCourses();

    @GET("api/modules/{id}")
    Call<ArrayList<Module>> getModuleList(@Path("id") int id);

    @GET("api/topics/{id}")
    Call<ArrayList<Topic>> getTopicList(@Path("id") int id);

    @POST("api/StudentAssignedCourses/{courseId}")
    Call<ResponseBody> registerCorper(@Path("courseId") int courseId,@Body CourseRegister courseRegister);

    @GET("api/topicmaterialuploads")
    Call<ArrayList<MaterialUpload>> getMaterialUpload(@Query("id") String topicID);

}

File: app/src/main/java/com/example/ogan/codedenim/gson/ApiMethods.kt
package com.example.ogan.codedenim.gson

import com.example.ogan.codedenim.courses.CourseRegister
import com.example.ogan.codedenim.user.Corp
import com.example.ogan.codedenim.user.Student
import com.example.ogan.codedenim.user.User
import okhttp3.ResponseBody
import retrofit2.Call
import retrofit2.http.*

interface ApiMethods {

    @get:GET("api/Courses")
    val courses: Call<ArrayList<Course>>

    @POST("api/accountapi/registercorper")
    fun registerCorper(@Body corp: Corp): Call<ResponseBody>

    @POST("api/AccountApi/RegisterUnderGraduate")
    fun registerStudent(@Body student: Student): Call<ResponseBody>

    @POST("api/AccountApi/OtherStudent")
    fun registerOtherStudent(@Body user: User): Call<ResponseBody>

    @FormUrlEncoded
    @POST("token")
    fun loginCorper(@Field("username") username: String, @Field("password") password: String, @Field("grant_type") grant_type: String): Call<ResponseBody>

    @GET("api/CourseCategories/CourseCategorys")
    fun getCategories(@Query("email") email: String?): Call<ArrayList<LearningPath>>

    @GET("api/Coursecategories/CourseCategory")
    fun getCourses(@Query("id") id: Int): Call<ArrayList<Course>>

    @GET("api/CourseCategories/MyCourses")
    fun getCourses(@Query("email") email: String?): Call<ArrayList<LearningPath>>

    @GET("api/modules/{id}")
    fun getModuleList(@Path("id") id: Int): Call<ArrayList<Module>>

    @GET("api/topics/{id}")
    fun getTopicList(@Path("id") id: Int): Call<ArrayList<Topic>>

    @POST("api/StudentAssignedCourses/{courseId}")
    fun registerStudent(@Path("courseId") courseId: Int, @Body courseRegister: CourseRegister): Call<ResponseBody>

    @GET("api/topicmaterialuploads")
    fun getMaterialUpload(@Query("id") topicID: String): Call<ArrayList<MaterialUpload>>

    @POST("api/CourseCategories/CheckPayment")
    fun checkPayment(@Body body: CheckPayment): Call<PaymentStatus>

    @GET("api/forumquestions")
    fun getForumQuestions(@Query("id") courseId: String?): Call<ArrayList<ForumQuestion>>

    @POST("api/forumquestions")
    fun postQuestion(@Query("email") email: String?, @Body body: ForumQuestion ): Call<ResponseBody>

    @GET("api/forumanswers")
    fun getAnswers(@Query("id") questionId: String?): Call<ArrayList<ForumAnswer>>

    @POST("api/forumanswers")
    fun postAnswer(@Query("email") email: String?, @Body body: ForumAnswer): Call<ResponseBody>

}
